### PR TITLE
refactor: 💡 healtcheck, deploy and controllers

### PR DESCRIPTION
**Pull request by punkbit**
## Why?

Amends required in healtcheck, such as preventing overriding home, on deployment make explicit controller setup.

## How?

- Small amend for healtcheck on home prefixing exec
- Deploy now includes controller setup
- Update docs

### CODE REVIEW

File: .scripts/dfx-identity.sh [hunk 1]
```diff
@@ -11,7 +12,15 @@ ALICE_PEM="$ALICE_HOME/.config/dfx/identity/default/identity.pem"
 BOB_PEM="$BOB_HOME/.config/dfx/identity/default/identity.pem"
 DEFAULT_PEM="$HOME/.config/dfx/identity/default/identity.pem"
```
Comment (ozwaldorf) [inline .scripts/dfx-identity.sh:13]: This should be 
`DEFAULT_PEM="$HOME/.config/dfx/identity/$(dfx identity whoami)/identity.pem"`
to allow any dfx identity to be used

File: .scripts/dfx-identity.sh [hunk 2]
```diff
@@ -11,7 +12,15 @@ ALICE_PEM="$ALICE_HOME/.config/dfx/identity/default/identity.pem"
...
\ No newline at end of file
+printf "🙋‍♀️ Identities\n\n"
+
+printf "👩🏽‍🦰 ALICE_PRINCIPAL_ID (%s)\n" "$ALICE_PRINCIPAL_ID"
+printf "👩🏽‍🦰 ALICE_HOME (%s)\n" "$ALICE_HOME"
+
+printf "👨🏽‍🦰 BOB_PRINCIPAL_ID (%s)\n" "$BOB_PRINCIPAL_ID"
+printf "👨🏽‍🦰 BOB_HOME (%s)\n" "$BOB_HOME"
+
+printf "👨🏾‍💻 DEFAULT_PRINCIPAL_ID (%s)\n" "$DEFAULT_PRINCIPAL_ID"
```
Comment (ozwaldorf) [inline .scripts/dfx-identity.sh:23]: Maybe add here:
`printf "👨🏾‍💻 DFX_IDENTITY(%s)\n" "$(dfx identity whoami)"`

File: .scripts/dip721/deploy-nft.sh [hunk 3]
```diff
@@ -17,6 +17,23 @@ if [[ "$MODE" == "reinstall" ]]; then
...
+
+# Note that the controller settings is important
+# you should have at least an identity you control
+# and the canister itself as controllers
+dfx canister --no-wallet \
+  --network "$NETWORK" \
+  update-settings \
+    --controller "$OWNER_PRINCIPAL_ID" \
+    --controller "$CREATED_NFT_CANISTER_ID" \
+  "$CREATED_NFT_CANISTER_ID"
```
Comment (ozwaldorf) [inline .scripts/dip721/deploy-nft.sh]: - We should wrap this with a check to see if the nft canister is already created
-  is the --controller necessary in the canister create? 
- read the canister id with --network flag to support local and ic deploys
```sh
CREATED_CANISTER_NFT_ID=$(dfx canister --network $NETWORK id $NFT)
if [[ ! $? -eq 0 ]]; then
    dfx canister --no-wallet create nft

    CREATED_NFT_CANISTER_ID=$(dfx canister --network $NETWORK id $NFT)

    printf "🤖 The created NFT Canster id is %s" "$CREATED_NFT_CANISTER_ID"

    # Note that the controller settings is important
    # you should have at least an identity you control
    # and the canister itself as controllers
    dfx canister --no-wallet \
      --network "$NETWORK" \
      update-settings \
      --controller "$OWNER_PRINCIPAL_ID" \
      --controller "$CREATED_NFT_CANISTER_ID" \
      "$CREATED_NFT_CANISTER_ID"
fi

**Review by ozwaldorf — CHANGES_REQUESTED** [review]

File: .scripts/dfx-identity.sh (hunk 1 above)
Comment (punkbit) [reply .scripts/dfx-identity.sh:13]: @ozwaldorf that can be quite useful, sounds very good! So, that'll be out of the scope for this PR, do you mind providing a PR on that, please? It'll require making other changes, which includes docs.

File: .scripts/dip721/deploy-nft.sh (hunk 3 above)
Comment (punkbit) [reply .scripts/dip721/deploy-nft.sh]: Nice one, I'll add the missing network!

So, the controller is really just being too verbose to communicate requirements, apart from that there's no other advantage.

For the wrapping, mind sending a PR on that? I think the use-case here, which is an implementation example, is already a bit too much.

Comment (punkbit) [reply .scripts/dip721/deploy-nft.sh]: @ozwaldorf updated, the remaining requests please address that separately. Confirm if everything's ok asap, thanks ;)

Comment (ozwaldorf) [reply .scripts/dip721/deploy-nft.sh]: @heldrida Everything else looks good, I will make a separate PR for the feature to use any identity. Use will end up exactly the same, but I will change the docs to reflect that the default identity is not required. IMHO the existing canister check should be included in this PR, as that doesn't pertain to the identity changes, and the controllers should only need to be set on the initial deploy.

Comment (punkbit) [reply .scripts/dip721/deploy-nft.sh]: @ozwaldorf fair enough, I'll see what I can do

Comment (punkbit) [reply .scripts/dip721/deploy-nft.sh]: @ozwaldorf updated!

**Review by ozwaldorf — APPROVED** [review]
Looks good!

**Pull request merged**